### PR TITLE
Upgarde to Vite v4

### PR DESCRIPTION
**Pull request by workshur**
- Vite v4
- SWC for vite/react 
- fixed specs/ got rid of warnings

**What changes did you make?** (Give an overview)

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [x] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: kafka-ui-react-app/src/components/Topics/Topic/Messages/MessagesTable.tsx [hunk 1]
```diff
@@ -24,70 +24,73 @@ const MessagesTable: React.FC = () => {
...
+          )}
+          {messages.length === 0 && !isFetching && (
+            <tr>
+              <td colSpan={10}>No messages found</td>
+            </tr>
+          )}
+        </tbody>
+      </Table>
+      {previewFor !== null && (
+        <PreviewModal
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/Topics/Topic/Messages/MessagesTable.tsx:83]: this exact change was implemented to aviod wrong nesting in the table , [here](https://github.com/provectus/kafka-ui/pull/3028/files) , with minor styling changes , cause it has to be at the top and the container must be relative.

**Review by Mgrdich — APPROVED** [review]

**Review by David-DB88 — COMMENTED** [review]
in test case  'should show preview modal with validation' have warning  in the MessagesTable.spec.tsx file 
validateDOMNesting(...): div cannot appear as a child of tr.

I think div in PreviewModal.styled.ts we should replace with th
or add th on PreviewModal Component

**Comment by David-DB88** [comment]
i will change it

File: kafka-ui-react-app/src/components/Topics/Topic/Messages/MessagesTable.tsx [hunk 2]
```diff
@@ -77,18 +77,19 @@ const MessagesTable: React.FC = () => {
...
-              <PreviewModal
-                values={previewFor === 'key' ? keyFilters : contentFilters}
-                toggleIsOpen={() => setPreviewFor(null)}
-                setFilters={(payload: PreviewFilter[]) =>
-                  previewFor === 'key'
-                    ? setKeyFilters(payload)
-                    : setContentFilters(payload)
-                }
-              />
-            )}
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Messages/MessagesTable.tsx:91]: @David-DB88 That's supposed to break the table. You can't just add a cell to a table without changing whole table html structure

**Pull request merged**